cmd/fabric8-jenkins-proxy/main_test.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/clients"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/configuration"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/storage"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/testutils"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/ory-am/dockertest.v3"
+	"os"
+	"testing"
+)
+
+const (
+	database = "tenant"
+	user     = "postgres"
+	password = "mysecretpassword"
+)
+
+func TestMain(m *testing.M) {
+	var db *sql.DB
+	var err error
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	resource, err := pool.Run("postgres", "9.6", []string{"POSTGRES_PASSWORD=" + password, "POSTGRES_DB=" + database})
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+
+	if err = pool.Retry(func() error {
+		var err error
+		db, err = sql.Open("postgres", fmt.Sprintf("postgres://%s:%s@localhost:%s/%s?sslmode=disable", user, password, resource.GetPort("5432/tcp"), database))
+		if err != nil {
+			return err
+		}
+		return db.Ping()
+	}); err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	// we needto make sure that the test can use the port exposed on the host
+	os.Setenv("JC_POSTGRES_PORT", resource.GetPort("5432/tcp"))
+	code := m.Run()
+
+	// When you're done, kill and remove the container
+	err = pool.Purge(resource)
+
+	os.Exit(code)
+}
+
+func TestProxy(t *testing.T) {
+	t.SkipNow()
```

Comment:
@vpavlin This one intrigues me. This test is supposed to be a replacement for _testrun.go_. Among other things, this test will actually create and start the test DB as part of the test. 
When I run this, I get:

    {"level":"fatal","msg":"Get /auth/realms/fabric8/: unsupported protocol scheme \"\"","time":"2018-01-23T13:45:30+01:00"}

I don't think I broke this. Any ideas? Shall we try to get this working end to end again? If you want to try, get the branch and uncomment this line and run the tests.

Reply:
This https://github.com/fabric8-services/fabric8-jenkins-proxy/pull/118/files#diff-951fb2cd29666d1232c48e2dd695609fR88 needs to resolve to https://github.com/hferentschik/fabric8-jenkins-proxy/blob/293291177020548dd5df52f5295eefd64e8a7207/cmd/fabric8-jenkins-proxy/main_test.go#L69 as in https://github.com/fabric8-services/fabric8-jenkins-proxy/blob/master/internal/testutils/testrun.go#L58

Reply:
thanks, I'll look into it.

Reply:
fixed